src/runtime/objmodel.cpp

```diff
@@ -490,6 +490,17 @@ BoxedDict* Box::getDict() {
 }
 
 Box* Box::getattr(const std::string& attr, GetattrRewriteArgs* rewrite_args) {
+
+    if (cls->tp_getattr) {
+        if (rewrite_args)
+            REWRITE_ABORTED("");
+
+        Box* rtn = cls->tp_getattr(this, const_cast<char*>(attr.c_str()));
+        if (rtn == NULL)
+            raiseExcHelper(AttributeError, "%s", attr.c_str());
+        return rtn;
+    }
+
```

Comment:
after we spoke I agreed, but after removing it I found a reason to keep it:

```
#0  pyston::Box::getattr (this=0x12700aa318, attr="register", rewrite_args=0x0) at src/runtime/objmodel.cpp:495
#1  0x0000000000847ba9 in pyston::getattrInternalGeneral (obj=0x12700aa318, attr="register", rewrite_args=0x0, cls_only=false, for_call=true, bind_obj_out=0x7fffffffcf38, r_bind_obj_out=0x7fffffffcf30) at src/runtime/objmodel.cpp:1290
#2  0x00000000008467f3 in callattrInternal (obj=0x12700aa318, attr=0x21994b0, scope=pyston::CLASS_OR_INST, rewrite_args=0x0, argspec=..., arg1=0x1270030508, arg2=0x0, arg3=0x0, args=0x0, keyword_names=0x7fffffffd330) at src/runtime/objmodel.cpp:2072
#3  0x000000000084bc94 in callattr (obj=0x12700aa318, attr=0x21994b0, flags=..., argspec=..., arg1=0x1270030508, arg2=0x0, arg3=0x0, args=0x0, keyword_names=0x7fffffffd330) at src/runtime/objmodel.cpp:2248
```

I could instead move the tp_getattr handling to callattr.